### PR TITLE
Create offline-registry-manipulation.yml

### LINKED ISSUE #211: offline registry manipulation with offreg.dll
originally suggested here: https://twitter.com/0gtweet/status/1344363283063701504?s=20

documentation:
- https://docs.microsoft.com/en-us/windows/win32/devnotes/offline-registry-library-portal
- https://docs.microsoft.com/en-us/windows/win32/devnotes/offline-registry-library-functions

### PR DESCRIPTION
**Pull request by johnk3r**
https://github.com/fireeye/capa-rules/issues/211

resolve #211

### CODE REVIEW

File: host-interaction/registry/offline-registry-manipulation.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    authOR: johnk3r
+    scope: file
+    att&ck:
+      - Discovery::Query Registry [T1012]
+    mbc:
+      - Operating System::Registry::Query Registry Value [C0036.006]
+      - Operating System::Registry::Create Registry Key [C0036.004]
+      - Operating System::Registry::Open Registry Key [C0036.003]
+    examples:
+      - 3edc06a4784d1038c999b984d558b60d:0xEE20
```
Comment (williballenthin) [inline host-interaction/registry/offline-registry-manipulation.yml:14]: is this the loaded virtual address?

**Comment by williballenthin** [comment]
#211 suggests that we create a rule to match on executables that *use* the offline registry routines, rather than implement them. the rule you've proposed here seems to match on `offreg.dll`, instead.

can you please change this rule to match on functions that call routines provided by `offreg.dll`?

**Review by williballenthin — CHANGES_REQUESTED** [review]
see discussion

**Comment by johnk3r** [comment]
The rule sample was the only malicious file I found with these routines. And the creation of the rule was based on the "finds" in that file.

I will dedicate myself to find a suitable sample and soon update this rule.

**Comment by recvfrom** [comment]
@johnk3r here's one: https://www.virustotal.com/gui/file/9a292192453b1746b23e26560d04bde422b44e48c1186e93ffb518fc8210225e/details

https://cisco.box.com/s/woy5cce2szd42gpf2yt67x5vhosgvlog (password: infected)

File: host-interaction/registry/offline-registry-manipulation.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    name: offline registry manipulation
+    namespace: host-interaction/registry
+    authOR: johnk3r
+    scope: function
+    att&ck:
+      - Discovery::Query Registry [T1012]
+    mbc:
+      - Operating System::Registry::Query Registry Value [C0036.006]
+      - Operating System::Registry::Create Registry Key [C0036.004]
+      - Operating System::Registry::Open Registry Key [C0036.003]
```
Comment (mr-tz) [inline host-interaction/registry/offline-registry-manipulation.yml:12]: I think we should split the apis and respective techniques create, query, open, etc. up into separate rules. Would you mind doing that? It's fine if we don't have samples for all and add those to the nursery.

Comment (johnk3r) [reply host-interaction/registry/offline-registry-manipulation.yml:12]: Good idea. Can I use the same sample if possible?

Comment (johnk3r) [reply host-interaction/registry/offline-registry-manipulation.yml]: Hello. Please evaluate the submitted rules. 

How can I do when a sample of a rule takes too long to process? As the pipeline does not accept samples that take a long time to process, should I create this rule as a nursery?

Comment (williballenthin) [reply host-interaction/registry/offline-registry-manipulation.yml]: yeah, putting the rule in the nursery is a fine choice.

File: host-interaction/registry/offline-set-registry.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+rule:
+  meta:
+    name: offline set registry
```
Comment (mr-tz) [inline host-interaction/registry/offline-set-registry.yml:3]:
```suggestion
    name: set registry value via offline registry library

```

**Review by mr-tz — CHANGES_REQUESTED** [review]
thanks for splitting these up!

**Comment by johnk3r** [comment]
Add nursery rule.

**Comment by mr-tz** [comment]
Hey @johnk3r, is this ready for another review? Looks like there's only minor formatting issues (line endings).

**Comment by johnk3r** [comment]
> Hey @johnk3r, is this ready for another review? Looks like there's only minor formatting issues (line endings).

Sorry man, I forgot to fix this.

**Comment by mr-tz** [comment]
tested locally and only got the following warning

```
$ python2 ../scripts/lint.py . -t offline --thorough

    (nursery) delete registry key via offline registry library
      WARN: missing examples: Add meta.examples so that the rule can be tested and verified
```

**Comment by mr-tz** [comment]
Thank you!

**Pull request merged**